modules/rgbd/src/hash_tsdf.cpp

```diff
@@ -817,5 +818,880 @@ int HashTSDFVolumeCPU::getVisibleBlocks(int currFrameId, int frameThreshold) con
     return numVisibleBlocks;
 }
 
+
+///////// GPU implementation /////////
+
+#ifdef HAVE_OPENCL
+
+HashTSDFVolumeGPU::HashTSDFVolumeGPU(float _voxelSize, const Matx44f& _pose, float _raycastStepFactor, float _truncDist, int _maxWeight,
+    float _truncateThreshold, int _volumeUnitRes, bool _zFirstMemOrder)
+    :HashTSDFVolume(_voxelSize, _pose, _raycastStepFactor, _truncDist, _maxWeight, _truncateThreshold, _volumeUnitRes, _zFirstMemOrder)
+{
+    int xdim, ydim, zdim;
+    if (zFirstMemOrder)
+    {
+        xdim = volumeUnitResolution * volumeUnitResolution;
+        ydim = volumeUnitResolution;
+        zdim = 1;
+    }
+    else
+    {
+        xdim = 1;
+        ydim = volumeUnitResolution;
+        zdim = volumeUnitResolution * volumeUnitResolution;
+    }
+    volStrides = Vec4i(xdim, ydim, zdim);
+
+    neighbourCoords = Vec8i(
+        volStrides.dot(Vec4i(0, 0, 0)),
+        volStrides.dot(Vec4i(0, 0, 1)),
+        volStrides.dot(Vec4i(0, 1, 0)),
+        volStrides.dot(Vec4i(0, 1, 1)),
+        volStrides.dot(Vec4i(1, 0, 0)),
+        volStrides.dot(Vec4i(1, 0, 1)),
+        volStrides.dot(Vec4i(1, 1, 0)),
+        volStrides.dot(Vec4i(1, 1, 1))
+    );
+
+    reset();
+}
+
+HashTSDFVolumeGPU::HashTSDFVolumeGPU(const VolumeParams & _params, bool _zFirstMemOrder)
+    : HashTSDFVolume(_params.voxelSize, _params.pose.matrix, _params.raycastStepFactor, _params.tsdfTruncDist, _params.maxWeight,
+        _params.depthTruncThreshold, _params.unitResolution, _zFirstMemOrder)
+{
+}
+// zero volume, leave rest params the same
+void HashTSDFVolumeGPU::reset()
+{
+    CV_TRACE_FUNCTION();
+    lastVolIndex = 0;
+    degree = 15;
+    buff_lvl = (int) pow(2, degree);
+    volUnitsData = cv::Mat(buff_lvl, volumeUnitResolution * volumeUnitResolution * volumeUnitResolution, rawType<TsdfVoxel>());
+    poses = cv::Mat(buff_lvl, 1, rawType<cv::Matx44f>());
+    lastVisibleIndexes = cv::Mat(buff_lvl, 1, CV_32S);
+    indexes = VolumesTable();
+    allVol2cam = cv::Mat(buff_lvl, 16, CV_32F);
+    allCam2vol = cv::Mat(buff_lvl, 16, CV_32F);
+}
+
```

Comment:
Why to have 2 functions with the same meaning?

Reply:
removed